### PR TITLE
configure initial routes and install CORS middleware

### PR DESCRIPTION
**Pull request by jonglass82**
The initial express routes have been set up and are prepared for our feature builds. I also implemented CORS middleware for Express because we're using a framework for our front end and will encounter CORS issues without it. 

I have tested each route in my browser and in Insomnia and they all return their expected response.

### CODE REVIEW

**Comment by heydanhey** [comment]
Looks good!
I'd recommend adding a `.gitignore` file and adding the line `/node_modules`. That dependencies don't need to be committed to the repo.

File: index.js [hunk 1]
```diff
@@ -1,23 +1,27 @@
...
+app.post('/api/create', (req,res) => {
+    res.send("your meme has been created!")
+    console.log("your meme has been created!")
+})
+
+app.put('/api/update/:id', (req,res) => {
+    res.send("your meme has been updated!")
+    console.log("your meme has been updated!")
+})
+
```
Comment (heydanhey) [inline index.js]: I know this is just a simple project - but when setting up the routes we can do it RESTful, just like in rails. A POST request to `/api/memes` should create a new meme. A GET to `/api/memes` should return a list of memes and a PUT to `/api/memes/:id` would be the update. What do you think?

Comment (jonglass82) [reply index.js]: Do you mean like editing the messages being sent when the routes are hit? Also if I make updates to my current branch and push it up again, does it just rewrite it? Or do I need to make a whole new branch?

Comment (heydanhey) [reply index.js]: Just the restful convention for the route paths - something like this:
```
app.get('/api/memes', (req, res) => {
    res.send('Here are all of your memes!');
});

app.post('/api/memes', (req, res) => {
    res.send("your meme has been created!")
})

app.put('/api/memes/:id', (req,res) => {
    res.send("your meme has been updated!")
})
```

You can commit and push right back to this branch.

Comment (jonglass82) [reply index.js]: Ok good call! Thanks Dan I made the  additional commit and pushed the branch back up.

**Pull request merged**